kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/internal/KmAnnotations.kt

```diff
@@ -34,6 +35,7 @@ internal fun KmAnnotation.toAnnotationSpec(): AnnotationSpec {
           addMember("%L = %L", name, arg.toCodeBlock())
         }
       }
+      .tag(this)
```

Comment:
This seemed like an oversight, most other specs are tagged with the metadata object.